script.lua

```diff
@@ -5861,9 +5869,11 @@ function setupPlayerArea(params)
         --Updates the number display
         selected.zone.editButton({index=0, label="Energy Cost: "..energy})
         for i, v in ipairs(elements) do
-            selected.elements[i].editButton({index=0, label=v})
+            if selected.elements[i] then
```

Comment:
in what situation did `selected.elements[i]` not exist?

Reply:
Currently, never. However, the Dark Fire implementation I've been using deletes one of the element bags, and sets its entry in `selected.elements` to `nil` to avoid storing a broken object reference. I'm not sure this is entirely necessary: I don't think calling `editButton()` on a broken object reference actually does any harm, so I could just not `nil` the entry out and it should work okay. So I can revert this change if you like. But I think it's fairly harmless to include, and the Dark Fire implementation I've been using seems somewhat safer with it.

Reply:
no this is harmless enough to leave it, was just wondering what the edge case you found